### PR TITLE
Feature/add satellites catalog app

### PR DESCRIPTION
**Pull request by smartsena**

### CODE REVIEW

File: catalog/filters.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        * end_date (str): filtering end date.
+    """
+    
+    satellite = rest_framework.NumberFilter(
+        field_name='satellite',
+        lookup_expr='exact',
+        required=True,
+    )
+
+    percent_cloud = rest_framework.NumberFilter(
```
Comment (dagnaldo) [inline catalog/filters.py:22]: what about `cloud_cover` only?

File: catalog/models.py [hunk 2]
```diff
@@ -38,4 +39,201 @@ def __str__(self) -> str:
         Returns:
             str: model data identifier.
         """
-        return self.name or self.identifier
\ No newline at end of file
+        return self.name or self.identifier
+
+
+class Catalogs (models.Model):
```
Comment (dagnaldo) [inline catalog/models.py:45]: Remove whitespace between class name and `(`

File: catalog/models.py [hunk 3]
```diff
@@ -38,4 +39,201 @@ def __str__(self) -> str:
...
+
+    )
+
+    max_native_zoom = models.IntegerField(
+        _('Maximum zoom scale'),
+        default=15,
+
+    )
+
+    co_cr = models.BigIntegerField(
```
Comment (dagnaldo) [inline catalog/models.py:111]: Is it necessary?

File: catalog/models.py [hunk 4]
```diff
@@ -38,4 +39,201 @@ def __str__(self) -> str:
...
+        _('Regional Coordenation code'),
+    )
+
+    ds_cr = models.CharField(
```
Comment (dagnaldo) [inline catalog/models.py:115]: Is it necessary?

File: catalog/models.py [hunk 5]
```diff
@@ -38,4 +39,201 @@ def __str__(self) -> str:
...
+        _('Regional Coordenation name'),
+        max_length=255,
+    )
+
+    co_funai = models.IntegerField(
```
Comment (dagnaldo) [inline catalog/models.py:120]: Is it necessary?

File: catalog/models.py [hunk 6]
```diff
@@ -38,4 +39,201 @@ def __str__(self) -> str:
...
+        _('Funai code'),
+    )
+
+    no_ti = models.CharField(
```
Comment (dagnaldo) [inline catalog/models.py:124]: Is it necessary?

File: catalog/models.py [hunk 7]
```diff
@@ -38,4 +39,201 @@ def __str__(self) -> str:
...
+        _('Name of Indigenous Lands'),
+        max_length=255,
+    )
+
+    municipio = models.CharField(
```
Comment (dagnaldo) [inline catalog/models.py:129]: Is it necessary?

File: catalog/models.py [hunk 8]
```diff
@@ -38,4 +39,201 @@ def __str__(self) -> str:
...
+        _('County'),
+        max_length=255,
+        blank=True,
+        null=True
+    )
+
+    sg_uf = models.CharField(
```
Comment (dagnaldo) [inline catalog/models.py:136]: Is it necessary?

File: catalog/models.py [hunk 9]
```diff
@@ -38,4 +39,201 @@ def __str__(self) -> str:
...
+        _('State aconymn'),
+        max_length=255,
+        blank=True,
+        null=True
+    )
+
+    resolution = models.IntegerField(
```
Comment (dagnaldo) [inline catalog/models.py:143]: Is it necessary?

File: catalog/models.py [hunk 10]
```diff
@@ -38,4 +39,201 @@ def __str__(self) -> str:
...
+        _('Spatial resolution'),
+        null=True,
+        blank=True,
+    )
+    
+    nu_latitude = models.CharField(
```
Comment (dagnaldo) [inline catalog/models.py:149]: Is it necessary?

File: catalog/models.py [hunk 11]
```diff
@@ -38,4 +39,201 @@ def __str__(self) -> str:
...
+        _('Numero latitude'),
+        max_length=255,
+        blank=True,
+        null=True
+    )
+
+    nu_longitude = models.CharField(
```
Comment (dagnaldo) [inline catalog/models.py:156]: Is it necessary?

File: catalog/models.py [hunk 12]
```diff
@@ -38,4 +39,201 @@ def __str__(self) -> str:
...
+    def __str__(self):
+        """Returns `catalog.Catalogs` string data.
+
+        Returns:
+            str: model data image.
+        """
+        return str(self.image)
+
+
+class Landsat8Catalog (Catalogs):
```
Comment (dagnaldo) [inline catalog/models.py:188]: Remove whitespace between classname and `(`

File: catalog/models.py [hunk 13]
```diff
@@ -38,4 +39,201 @@ def __str__(self) -> str:
...
+    )
+
+    class Meta:
+        """"Meta class for `catalog.Landsat8Catalog` model."""
+        app_label = 'catalog'
+        verbose_name = 'Landsat8 Catalog'
+        verbose_name_plural = 'Landsat8 Catalogs'
+
+
+class Sentinel2Catalog (Catalogs):
```
Comment (dagnaldo) [inline catalog/models.py:212]: Remove whitespace between classname and (

File: catalog/views.py [hunk 14]
```diff
@@ -18,4 +24,57 @@ class SatelliteView(AuthModelMixIn, generics.ListAPIView):
...
+    """
+
+    bbox_filter_field = 'geom'
+    filterset_class = catalog_filters.CatalogFilter
+    filter_backends = (
+        DjangoFilterBackend,
+        gis_filters.InBBoxFilter,
+    )
+
+    sat_landsat8=2
```
Comment (dagnaldo) [inline catalog/views.py:57]: what about `landsat8_id`?

File: catalog/views.py [hunk 15]
```diff
@@ -18,4 +24,57 @@ class SatelliteView(AuthModelMixIn, generics.ListAPIView):
...
+    sat_sentinel2=3
```
Comment (dagnaldo) [inline catalog/views.py:58]: what about `sentinel2_id`?

File: catalog/views.py [hunk 16]
```diff
@@ -18,4 +24,57 @@ class SatelliteView(AuthModelMixIn, generics.ListAPIView):
...
+
+    def get_queryset(self):
+        request_satellite = int(self.request.GET.get('satellite'))
+        if request_satellite == self.sat_sentinel2:
+            print("sentinel2     ---->")
```
Comment (dagnaldo) [inline catalog/views.py:63]: Please, remove unecessary prints.

File: catalog/views.py [hunk 17]
```diff
@@ -18,4 +24,57 @@ class SatelliteView(AuthModelMixIn, generics.ListAPIView):
...
+            return models.Sentinel2Catalog.objects.all()
+        elif request_satellite == self.sat_landsat8:
+            print("landsat8     ---->")
```
Comment (dagnaldo) [inline catalog/views.py:66]: Please, remove unecessary prints.

File: catalog/views.py [hunk 18]
```diff
@@ -18,4 +24,57 @@ class SatelliteView(AuthModelMixIn, generics.ListAPIView):
...
+            return models.Landsat8Catalog.objects.all()
+
+    def get_serializer_class(self):
+        request_satellite = int(self.request.GET.get('satellite'))
+        if request_satellite == self.sat_sentinel2:
+            print("sentinel2")
```
Comment (dagnaldo) [inline catalog/views.py:72]: Please, remove unecessary prints.

File: catalog/views.py [hunk 19]
```diff
@@ -18,4 +24,57 @@ class SatelliteView(AuthModelMixIn, generics.ListAPIView):
...
+            return serializers.Sentinel2CatalogSerializer
+        elif request_satellite == self.sat_landsat8:
+            print("landsat8")
```
Comment (dagnaldo) [inline catalog/views.py:75]: Please, remove unecessary prints.

File: cmr2_api/settings.py [hunk 20]
```diff
@@ -45,6 +45,7 @@
     'rest_framework_gis',
     'django_filters',
     'corsheaders',
+    'django.contrib.gis',
```
Comment (dagnaldo) [inline cmr2_api/settings.py:48]: Check position, it might be on `django.contrib...` imports.

**Review by dagnaldo — CHANGES_REQUESTED** [review]

File: catalog/filters.py (hunk 1 above)
Comment (smartsena) [reply catalog/filters.py:22]: Chenged by cloud_cover.

File: catalog/models.py (hunk 2 above)
Comment (smartsena) [reply catalog/models.py:45]: Removed.

File: catalog/models.py (hunk 4 above)
Comment (smartsena) [reply catalog/models.py:115]: This field can be removed after the relationship of this "APP Catalog" with the "APP Funai". In the meantime the "ds_cr" attribute is required.

File: catalog/models.py (hunk 3 above)
Comment (smartsena) [reply catalog/models.py:111]: this field can be removed after the relationship of this "APP Catalog" with the "APP Funai". In the meantime the "ds_cr" attribute is required. 

Under study at CARD: https://xskylab.atlassian.net/browse/CMR-127

File: catalog/models.py (hunk 12 above)
Comment (smartsena) [reply catalog/models.py:188]: Removed.

File: catalog/models.py (hunk 6 above)
Comment (smartsena) [reply catalog/models.py:124]: For this demand it is necessary.
Under study at CARD: https://xskylab.atlassian.net/browse/CMR-127

File: catalog/models.py (hunk 7 above)
Comment (smartsena) [reply catalog/models.py:129]: For this demand it is necessary.
Under study at CARD: https://xskylab.atlassian.net/browse/CMR-127

File: catalog/models.py (hunk 8 above)
Comment (smartsena) [reply catalog/models.py:136]: For this demand it is necessary.
Under study at CARD: https://xskylab.atlassian.net/browse/CMR-127

File: catalog/models.py (hunk 9 above)
Comment (smartsena) [reply catalog/models.py:143]: For this demand it is necessary.
Under study at CARD: https://xskylab.atlassian.net/browse/CMR-127

File: catalog/models.py (hunk 10 above)
Comment (smartsena) [reply catalog/models.py:149]: For this demand it is necessary.
Under study at CARD: https://xskylab.atlassian.net/browse/CMR-127

File: catalog/models.py (hunk 13 above)
Comment (smartsena) [reply catalog/models.py:212]: Removed.

File: catalog/views.py (hunk 18 above)
Comment (smartsena) [reply catalog/views.py]: Removed.

File: catalog/views.py (hunk 19 above)
Comment (smartsena) [reply catalog/views.py]: Removed.

File: catalog/views.py (hunk 17 above)
Comment (smartsena) [reply catalog/views.py]: Removed.

File: catalog/views.py (hunk 14 above)
Comment (smartsena) [reply catalog/views.py:57]: Adjusted.

File: catalog/views.py (hunk 15 above)
Comment (smartsena) [reply catalog/views.py:58]: Adjusted.

File: cmr2_api/settings.py (hunk 20 above)
Comment (smartsena) [reply cmr2_api/settings.py:48]: Adjusted.

**Pull request merged**